sendgrid/sendgrid.go

```diff
@@ -12,12 +14,46 @@ type Sendgrid struct {
 
 func (sg *Sendgrid) GetRoutes() []*rest.Route {
 	return []*rest.Route{
-		// rest.Head("/sendgrid", headHandler),
 		rest.Post("/sendgrid/mx", sg.ReceiveHandler),
 	}
 }
 
 func (sg *Sendgrid) ReceiveHandler(w rest.ResponseWriter, r *rest.Request) {
-	w.WriteHeader(500)
-	w.WriteJson(struct{ Error string }{Error: "not implemented"})
+	fmt.Printf("POST to '%s': %#v\n\n", r.URL.String(), r)
+
+	r.Body = http.MaxBytesReader(w.(http.ResponseWriter), r.Body, 1024*1024*50)
+	defer r.Body.Close()
+	r.ParseMultipartForm(64 << 20)
+
+	form := r.PostForm
+	// for k, v := range form {
+	// 	fmt.Printf("data %s: %s \n", k, v)
+	// }
+
+	recipient := form.Get("to")
+	re, _ := regexp.Compile("<(.*?)>")
```

Comment:
The docs says the To header is just taken from the headers, so using https://pkg.go.dev/net/mail#ParseAddress would be better. (Or ParseAddressList)?

Actually -- I think you should be using `envelope.to` instead (the envelope recipient instead of whatever is in the headers).

Reply:
altered the code.... I'm no Go developer though so maybe I could have done this better? Let me know what you think